### ISSUE TITLE
Classifier for ListQA

### ISSUE THREAD

**Issue by jdpsen** [primeqa/primeqa#87]
_jaydesen created the following on Mar 08:_  

 Aim is to develop a classifier to detect questions that need list answers.  

_Original issue: https://github.ibm.com/ai-foundation/PrimeQA/issues/53_

**Comment by jdpsen** [comment]
_On Mar 08, sjrosenthal commented:_
The classifier is given the question and the label as collection or other. Other is the GooAQ questions which have a short answer.

**GooAQ**
Training Data: 612983 (323734 collection, 289265 other)
Dev Data: 4661
Test Data: 4657

**GooAQ F1 collection/other**	
mBERT 0.995/0.989
xlm-roberta 0.995/0.990
xlm-roberta-bal  0.996/0.991

**Comment by jdpsen** [comment]
_On Mar 08, jaydesen commented:_
Need to check on other datasets : NQ and others ?
CCQA- not well labelled  for list questions.

**Comment by jdpsen** [comment]
_On Mar 14, jaydesen commented:_
@sjrosenthal : can you please paste the results here which we were discussing on this last Tuesday

**Comment by jdpsen** [comment]
_On Mar 14, sjrosenthal commented:_
GooAQ does not transfer well to NQ Dev

**NQ** 

Dev Data: 3397 (73 collection, 3324 other): 

**NQ F1 Collection/Other**
xlm-roberta 0.071/0.841
xlm-roberta-bal 0.071/0.868

I expect these are because the questions are partly different formats (because GooAQ is search queries). It also may be due to size, I will explore this by experimenting with a larger amount of NQ list data using part of the training set.

**Comment by jdpsen** [comment]
_On Mar 18, sjrosenthal commented:_
more balanced data: 

    NQ Train: Collection = 2620, Other = 3000

    Collection (0) Other (1)
    F1: 0.44129851 0.67954641

Based on exploring examples we actually are doing better than it appears - and in fact many of the errors are in mislabeling highlighting the need for the annotation task to confirm they are indeed list questions:

```
"-9219200922296253160": {
        "conf": "3.398342",
        "label": 0,
        "language": "english",
        "pred": "collection",
        "sentence1": "who was born on the 15th of march"
    },
    "-9209533347176381800": {
        "conf": "4.632435",
        "label": 0,
        "language": "english",
        "pred": "collection",
        "sentence1": "protocols used in physical layer of osi model"
    },
   "-9195490722712249555": {
        "conf": "2.8752418",
        "label": 0,
        "language": "english",
        "pred": "other",
        "sentence1": "who played in the movie dirty rotten scoundrels"
    },
    "-9195137870836965211": {
        "conf": "3.7168434",
        "label": 0,
        "language": "english",
        "pred": "collection",
        "sentence1": "how do you spell stacy for a girl"
    },
    "-9194407123608287683": {
        "conf": "4.343223",
        "label": 0,
        "language": "english",
        "pred": "other",
        "sentence1": "how long is a normal high school football game"
    },
    "-9191982074433010777": {
        "conf": "5.779284",
        "label": 0,
        "language": "english",
        "pred": "collection",
        "sentence1": "what were the main causes of the french revolution"
    },
    "-9191026149096320471": {
        "conf": "4.334317",
        "label": 1,
        "language": "english",
        "pred": "collection",
        "sentence1": "list of inclusive schools from department of school education"
    },
```

**Comment by jdpsen** [comment]
_On Mar 18, sjrosenthal commented:_
I've confirmed that running the full NQ dataset (all question types) on the model fine-tuned for lists does not perform well, indicating a list qa type classifer is needed: 

     F1 28.75% P / F1 18.67% M 

I've also confirmed that there are many lists in a Wikipedia page because the TOC and part of the info boxes are lists.

**Comment by jdpsen** [comment]
_On Mar 24, jaydesen commented:_
LIstQA model tends to pick longer spans for non-list questions. 
Waiting for better annotated data to train a robust model. 
Release plan: GooAQ subset for list questions and existing QA model fine-tuned on GooAQ list qs.

**Comment by jdpsen** [comment]
@seirasto : are there any next experiments you are trying related to this ? Or we can close this ?

**Comment by jdpsen** [comment]
being tracked internally here: https://github.ibm.com/IBM-Research-AI/PrimeQA-Research/issues/7
